src/query/storages/cache/src/providers/file_cache.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_exception::Result;
+use opendal::Object;
+
+use crate::CacheSettings;
+use crate::ObjectCacheProvider;
+
+// TODO(bohu): local disk file based LRU cache
+pub struct FileCache {}
```

Comment:
In the past, common-cache included a disk-cache implementation. Removed in #8166, perhaps we could consider taking it back.

Reply:
For file cache, there are many to consider, for example: if we have a range cached: <location, 100, 150>, if a request is <location, 101, 140>, we should consider the overlap.